Snowplow/Internal/Session/SPSession.m

```diff
@@ -145,20 +139,21 @@ - (void) setBackgroundTimeout:(NSInteger)backgroundTimeout {
 }
 
 - (NSDictionary *) getSessionDictWithEventId:(NSString *)eventId {
-    NSMutableDictionary *result;
-    if (!_isSessionCheckerEnabled) {
-        result = [_sessionDict mutableCopy];
-    } else {
-        @synchronized (self) {
+    @synchronized (self) {
+        if (_isSessionCheckerEnabled) {
             if ([self shouldUpdateSession]) {
                 [self updateSessionWithEventId:eventId];
+                if (self.onSessionStateUpdate) {
+                    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+                        self.onSessionStateUpdate(self.state);
+                    });
+                }
             }
             self.lastSessionCheck = [SPUtilities getTimestamp];
-            result = [_sessionDict mutableCopy];
         }
+        _state.userId = _userId;
```

Comment:
No, it can't change. It's that because for the version 2 there was the intention to allow the creation of an InstallationID (alias of session_userID) available also when the session is not enabled. On mobile, the session and the userID (actually it's a device ID) don't necessarily needs to stay together. Unfortunately, I cut back the changes it never saw the light but it was planned in the RFC v2.